src/main/java/DilemmaDetector/Consequences/CollisionConsequencePredictor.java

```diff
@@ -0,0 +1,152 @@
+package DilemmaDetector.Consequences;
+
+import DilemmaDetector.Simulator.Actor;
+import DilemmaDetector.Simulator.PhysicsUtils;
+import generator.Model;
+import project.Decision;
+import project.Living_entity;
+import project.MyFactory;
+import project.Vehicle;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CollisionConsequencePredictor {
+    private IConsequenceContainer consequenceContainer;
+    private MyFactory factory;
+    private Model model;
+
+    public CollisionConsequencePredictor(IConsequenceContainer consequenceContainer, MyFactory factory, Model model) {
+        this.consequenceContainer = consequenceContainer;
+        this.factory = factory;
+        this.model = model;
+    }
+
+    public void createCollisionConsequences(Decision decision, Actor victimActor, Actor other) {
+        List<Living_entity> individualVictims = getLivingEntitiesFromActor(victimActor);
+
+        for (Living_entity victim : individualVictims) {
+            ConsequenceType consequenceType = getHealthConsequenceType(victimActor, other);
+            if (consequenceType != ConsequenceType.NO_CONSEQUENCE) {
+                consequenceContainer.addHealthConsequence(decision, victim, consequenceType);
+            }
+        }
+    }
+
+    private ConsequenceType getHealthConsequenceType (Actor victimActor, Actor other){
+        double speedOfCollision = PhysicsUtils
+                .GetRelativeSpeed(other.getRigidBody().getSpeed(), victimActor.getRigidBody().getSpeed())
+                .getMagnitude();
+
+        double killProbability = fatalInjuryProbability(speedOfCollision);
+        double severInjuryProbability = severInjuryProbability(speedOfCollision);
+        double lightInjuryProbability = minorInjuryProbability(speedOfCollision);
+        double maxProbability = killProbability;
+        if (maxProbability < severInjuryProbability)
+            maxProbability = severInjuryProbability;
```

Comment:
Ta logika będzie do poprawy jak będziemy rozróżniać różne konsekwencje dla różnych ludzi w aucie. Tutaj można wydzielić jakieś zadanie, żeby ładnie to przemyśleć i zaimplementować.